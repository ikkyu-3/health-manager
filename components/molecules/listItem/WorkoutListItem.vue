<template>
  <v-list-item>
    <v-list-item-action>
      <select-button :is-active="isActive" :text="index" @click="click" />
    </v-list-item-action>

    <v-list-item-content>
      <v-list-item-title>{{ name }}</v-list-item-title>
    </v-list-item-content>
  </v-list-item>
</template>

<script lang="ts">
import { createComponent, computed } from '@vue/composition-api'
import SelectButton from '@/components/molecules/button/SelectButton.vue'

export default createComponent({
  components: { SelectButton },
  props: {
    name: String,
    index: String,
    click: Function
  },
  setup(props) {
    const isActive = computed(() => !!props.index)
    return { isActive }
  }
})
</script>
