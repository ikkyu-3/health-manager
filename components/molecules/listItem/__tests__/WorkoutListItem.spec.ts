import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import WorkoutListItem from '@/components/molecules/listItem/WorkoutListItem.vue'

const options = {
  localVue: createLocalVue(),
  vuetify: new Vuetify()
}

describe('molecules/listItem/WorkoutListItem.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(WorkoutListItem, {
      ...options,
      propsData: { name: 'Test', index: '1' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('nameで指定した文字列が表示される', () => {
    const wrapper = mount(WorkoutListItem, {
      ...options,
      propsData: { name: 'Test', index: '1' }
    })
    expect(wrapper.find('.v-list-item__title').text()).toBe('Test')
  })

  it('indexを指定した場合、アクティブなアイコンが表示される', () => {
    const wrapper = mount(WorkoutListItem, {
      ...options,
      propsData: { name: 'Test', index: '1' }
    })
    expect(wrapper.find('.icon').classes()).toEqual(
      expect.arrayContaining(['active'])
    )
  })

  it('indexを指定しない場合、アクティブでないアイコンが表示される', () => {
    const wrapper = mount(WorkoutListItem, {
      ...options,
      propsData: { name: 'Test', index: '' }
    })
    expect(wrapper.find('.icon').classes()).not.toEqual(
      expect.arrayContaining(['active'])
    )
  })
})
