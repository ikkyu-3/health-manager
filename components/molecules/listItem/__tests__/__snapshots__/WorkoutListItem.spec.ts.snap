// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/listItem/WorkoutListItem.vue snapshot 1`] = `"<div tabindex=\\"-1\\" class=\\"v-list-item theme--light\\"><div class=\\"v-list-item__action\\"><div class=\\"icon active\\"><span>1</span></div></div> <div class=\\"v-list-item__content\\"><div class=\\"v-list-item__title\\">Test</div></div></div>"`;
